server/src/main/java/org/apache/druid/server/http/DataSourcesResource.java

```diff
@@ -549,9 +549,7 @@ private ImmutableDruidDataSource getDataSource(final String dataSourceName)
         continue;
       }
 
-      if (!tierDistinctSegments.containsKey(tier)) {
-        tierDistinctSegments.put(tier, new HashSet<>());
-      }
+      tierDistinctSegments.putIfAbsent(tier, new HashSet<>());
```

Comment:
This is a good candidate for `computeIfAbsent`, to avoid the extra allocation. (It's short lived so not really a big deal, but, still.)

Reply:
Done.